pcdsdevices/targets.py

```diff
@@ -260,3 +264,210 @@ def move(self, nxspaces, nyspaces, wait=False):
+
+    def get_presets(self):
+        """
+        Get the saved presets if any.
+
+        Returns
+        -------
+        pos : tuple
+            Three coordinate positions - bottom_left, top_left, top_right
+        """
+        try:
+            pos = [self.x.presets.positions.x_bottom_left.pos,
+                   self.y.presets.positions.y_bottom_left.pos,
+                   self.x.presets.positions.x_top_left.pos,
+                   self.y.presets.positions.y_top_left.pos,
+                   self.x.presets.positions.x_top_right.pos,
+                   self.y.presets.positions.y_top_right.pos]
+            bottom_left = (pos[0], pos[1])
+            top_left = (pos[2], pos[3])
+            top_right = (pos[4], pos[5])
+            return bottom_left, top_left, top_right
+        except Exception:
+            logger.warning('Could not get presets, try to set_presets.')
+
+    def map_points(self, snake_like=True, show_grid=False):
+        """
+        Map all the sample positions in 2-d coordinates.
+
+        Parameters
+        ----------
+        snake_like : bool
+            Indicates if the points should be mapped in a snake-like pattern.
+        show_grid : bool
+            Indicates if the grid of mapped points should be displayed.
+
+        Returns
+        -------
+        coord : list
+            List of all coordinate points for samples on the grid.
+        """
+        bottom_left, top_left, top_right = self.get_presets()
+        if all([bottom_left, top_left, top_right]) is None:
+            msg = 'Could not get presets, make sure you set presets'
+            logger.error(msg)
+            raise ValueError(msg)
+
+        # leaving these guys here for reference only for now
+        # # distance from bottom_left to top_left
+        # height = np.sqrt(np.power((top_left[0] - bottom_left[0]), 2)
+        #                  + np.power((top_left[1] - bottom_left[1]), 2))
+        # # distance from top_left to top_right
+        # width = np.sqrt(np.power((top_right[0] - top_left[0]), 2)
+        #                 + np.power((top_right[1] - top_left[1]), 2))
+
+        x_grid_points = np.arange(top_left[0], top_right[0], self.x_spacing)
+        y_grid_points = np.arange(bottom_left[1], top_left[1], self.y_spacing)
+
+        # The meshgrid function returns
+        # two 2-dimensional arrays
+        xx, yy = np.meshgrid(x_grid_points, y_grid_points,
+                             sparse=False, indexing='xy')
+
+        if show_grid:
+            plt.plot(xx, yy, marker='.', color='k', linestyle='none')
+            plt.show()
+
+        # flat out the arrays of points
+        if not snake_like:
+            flat_xx = list(chain.from_iterable(xx))
+            flat_yy = list(chain.from_iterable(yy))
+            return flat_xx, flat_yy
+
+        xx = self.snake_grid_list(xx)
+        yy = self.snake_grid_list(yy)
+        return xx, yy
+
+    def map_skewed(self, height=2500, width=2500, snake_like=True,
```

Comment:
This is definitely not how I should handle the affine and projective transformation, but I just have this here for now for testing ... not even sure if it will help at all or if it is correct
This is just a little skew adjustment, that's all.... 
From a grid like this:
![skewed_grid](https://user-images.githubusercontent.com/62306310/102533252-04669c80-405a-11eb-9c73-4e2b83a2f760.PNG)

![skewed_plan](https://user-images.githubusercontent.com/62306310/102533289-0f213180-405a-11eb-8d56-806c55690e41.PNG)